### PR TITLE
`Development`: Speedup client tests for button component

### PR DESCRIPTION
**Pull request by MerlinMehmed**
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x]  I ran the tests locally and all tests passed.

### Motivation and Context

The button component had slow test execution. The runtime locally was 17.2 in the beginning which was decreased to ~4 seconds

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

File: src/test/javascript/spec/component/shared/button.component.spec.ts [hunk 1]
```diff
@@ -3,32 +3,32 @@ import { By } from '@angular/platform-browser';
...
     let fixture: ComponentFixture<ButtonComponent>;
     let debugElement: DebugElement;
 
-    let clickSpy: SinonSpy;
+    let clickSpy: jest.SpyInstance;
 
     beforeEach(async () => {
         return TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, ArtemisSharedComponentModule],
+            imports: [TranslateModule.forRoot(), ArtemisTestModule],
```
Comment (lschlesinger) [inline src/test/javascript/spec/component/shared/button.component.spec.ts:28]: You can remove the `TranslateModule.forRoot()`, I just checked it, tests run without it as well ;) 
```suggestion
            imports: [ArtemisTestModule],
```

**Review by lschlesinger — CHANGES_REQUESTED** [review]
For all the `async () ...await`s in the test cases, I'd suggest replacing `async` by `fakeAsync` and `await fixture.whenStable();` by `tick()`. I checked for every test case, it works. And it does better fit our testing guidelines I think ;)

**Review by lschlesinger — DISMISSED** [review]
Looks good, thanks for adding the suggestions 👍

File: src/test/javascript/spec/component/shared/button.component.spec.ts [hunk 2]
```diff
@@ -1,34 +1,33 @@
...
 
 describe('ButtonComponent', () => {
     let comp: ButtonComponent;
     let fixture: ComponentFixture<ButtonComponent>;
     let debugElement: DebugElement;
 
-    let clickSpy: SinonSpy;
+    let clickSpy: jest.SpyInstance;
 
     beforeEach(async () => {
```
Comment (imstreetspirit) [inline src/test/javascript/spec/component/shared/button.component.spec.ts:25]: I think you can remove `async` here

File: src/test/javascript/spec/component/shared/button.component.spec.ts [hunk 3]
```diff
@@ -1,34 +1,33 @@
...
         return TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, ArtemisSharedComponentModule],
+            imports: [ArtemisTestModule],
+            declarations: [ButtonComponent, FeatureToggleDirective, TranslatePipeMock, MockDirective(NgbTooltip), MockDirective(TranslateDirective)],
```
Comment (imstreetspirit) [inline src/test/javascript/spec/component/shared/button.component.spec.ts:28]: can `FeatureToggleDirective` be mocked?

**Review by imstreetspirit — COMMENTED** [review]
i have only two small comments,  besides it looks good to me.

File: src/test/javascript/spec/component/shared/button.component.spec.ts (hunk 3 above)
Comment (MerlinMehmed) [reply src/test/javascript/spec/component/shared/button.component.spec.ts:28]: No, it is needed for the tests which test the disabled attribute of the button

**Review by imstreetspirit — DISMISSED** [review]
thanks for the clarification, the changes look good to me.

File: src/test/javascript/spec/component/shared/button.component.spec.ts [hunk 4]
```diff
@@ -1,34 +1,33 @@
...
+    let clickSpy: jest.SpyInstance;
 
-    beforeEach(async () => {
+    beforeEach(() => {
         return TestBed.configureTestingModule({
-            imports: [TranslateModule.forRoot(), ArtemisTestModule, ArtemisSharedComponentModule],
+            imports: [ArtemisTestModule],
+            declarations: [ButtonComponent, FeatureToggleDirective, TranslatePipeMock, MockDirective(NgbTooltip), MockDirective(TranslateDirective)],
             providers: [
-                JhiLanguageHelper,
```
Comment (MDK5) [inline src/test/javascript/spec/component/shared/button.component.spec.ts:31]: The button component has no providers in its constructor, so normally no provider mocks should be needed. Maybe the FeatureToggleService needs to be mocked. Did you make sure all of these are needed?

File: src/test/javascript/spec/component/shared/button.component.spec.ts [hunk 5]
```diff
@@ -101,63 +100,63 @@ describe('ButtonComponent', () => {
...
-        expect(button.disabled).to.be.true;
+        expect(button).not.toBe(null);
+        expect(button.disabled).toBe(true);
         button.click();
-        await fixture.whenStable();
+        tick();
 
-        expect(clickSpy).to.not.have.been.called;
-    });
+        expect(clickSpy).toHaveBeenCalledTimes(0);
```
Comment (MDK5) [inline src/test/javascript/spec/component/shared/button.component.spec.ts:125]:
```suggestion
        expect(clickSpy).not.toHaveBeenCalled();
```
As by out guidelines

**Review by MDK5 — CHANGES_REQUESTED** [review]
Reviewed code, left two comments below

File: src/test/javascript/spec/component/shared/button.component.spec.ts (hunk 5 above)
Comment (MerlinMehmed) [reply src/test/javascript/spec/component/shared/button.component.spec.ts:125]: I've missed that, thanks

File: src/test/javascript/spec/component/shared/button.component.spec.ts (hunk 4 above)
Comment (MerlinMehmed) [reply src/test/javascript/spec/component/shared/button.component.spec.ts:31]: You are right, only the FeatureToggleService needs to be mocked, I removed the rest

**Review by Malyuk-A — CHANGES_REQUESTED** [review]
Tests work and are quite fast now (locally for me ~600 ms) 👍,
Code changes are good but I think you forgot to remove the unused imports:
image
image

**Review by imstreetspirit — APPROVED** [review]
looks good to me

**Review by MDK5 — APPROVED** [review]
Code changes look good to me now

**Review by Malyuk-A — APPROVED** [review]
Unused imports are removed. Good job 👍

**Review by jhainzlschmid — APPROVED** [review]
Changes look good to me 👍

**Pull request merged**